lucene/queries/src/java/org/apache/lucene/queries/function/valuesource/vectors/FloatVectorFieldSource.java

```diff
@@ -0,0 +1,213 @@
+    BFLOAT16 // Compressed float[] to bfloat. Separator = "|"
+  }
+
+  public static float[] rawStringToVector(String stringifiedVector) {
+    //format: 1.0,2.345
+
+    String[] split = stringifiedVector.split(",");
+    float[] output = new float[split.length];
+
+    for(int i=0;  i< split.length; i++) {
+      output[i] = Float.parseFloat(split[i]);
+    }
+
+    return output;
+  }
+
+  public static String vectorToRawString(float[] vector) {
+    StringBuilder sb = new StringBuilder();
+    for (int i = 0; i < vector.length; i++) {
+      if (sb.length() > 0) {
+        sb.append(',');
+      }
+      sb.append(Float.toString(vector[i]));
+    }
+    return sb.toString();
+  }
+
+  public static BytesRef vectorToRawStringBytesRef(float[] vector){
+      BytesRefBuilder builder = new BytesRefBuilder();
+      byte[] bytes;
+      for (int i=0; i<vector.length; i++){
+        if (builder.length() > 0){
+          builder.append((byte)',');
+        }
+        bytes = Float.toString(vector[i]).getBytes(StandardCharsets.UTF_8);
+        builder.append(bytes, 0, bytes.length);
+      }
+      return builder.get();
+  }
+
+  @Override
+  public FunctionValues getValues(Map context, LeafReaderContext readerContext) throws IOException {
+
+      final BinaryDocValues binaryValues = DocValues.getBinary(readerContext.reader(), field);
+      return new FunctionValues() {
+        int lastDocID = -1;
+
+        private BytesRef getValueForDoc(int doc) throws IOException {
+          if (doc < lastDocID) {
+            throw new IllegalArgumentException("docs were sent out-of-order: lastDocID=" + lastDocID + " vs docID=" + doc);
+          }
+          lastDocID = doc;
+          int curDocID = binaryValues.docID();
+          if (doc > curDocID) {
+            curDocID = binaryValues.advance(doc);
+          }
+          if (doc == curDocID) {
+            return decode(binaryValues.binaryValue());
+          } else {
+            return null;
+          }
+        }
+
+
+        //TODO: maybe change to float[][] later... ?
+        protected BytesRef decode(BytesRef encoded){
+          byte[] encodedVectors = encoded.bytes;
+          BytesRefBuilder currentEncodedVector = new BytesRefBuilder();
+          BytesRefBuilder decodedVectorString = new BytesRefBuilder();  //TODO: float[][] later
+
+          for (int i=encoded.offset; i< encoded.offset + encoded.length; i++) {
+            if (encodedVectors[i] == (byte) '|' || i == encoded.offset + encoded.length - 1) {
+              float[] currentVector = null;
+              switch (encoding) {
+                case BFLOAT16:
+                  //TODO
+                  break;
+                case BASE64:
+                  currentVector = base64ToVector(currentEncodedVector.toBytesRef().bytes);
+                  break;
+                case STRING:
+                  currentVector = rawStringToVector(currentEncodedVector.get().toString());
```

Comment:
Should be
```suggestion
                  currentVector = rawStringToVector(new String(currentEncodedVector.get().bytes, StandardCharsets.UTF_8));
```
otherwise not work